### DISCUSSION TITLE
How to create a custom icon set for DACAI screens

### DISCUSSION THREAD

**Discussion by daniel-volz** [mriscoc/Ender3V2S1#112] [Q&A] [answered]
Hi @all,

How are custom icons created?

Do you have an entry point for me, I want to improve the current icons.
I´m having a Ender 3 S1 and DACAI-Screen and can´t figure out how to decompile this 9.zico file.

The python scripts I found seems to convert only 9.ico files for DWIN devices.

Many Thanks in advance.

Daniel

**Comment by mriscoc** [comment, marked as answer]
Hi, to create the icons set for DACAI screens you need to use some Chinese tools from: http://www.gz-dc.com/category/typeid/412#mainTop  
  
Download and use Projectconvert.exe from the xgus tool, that allows to convert a DWIN icon set source project to DACAI: [xgus转换工具.zip](https://github.com/mriscoc/Ender3V2S1/files/10182111/xgus.zip)  
  
![ProjectConvert](https://user-images.githubusercontent.com/2745567/206357265-e2c7a48b-f2b4-473e-83c3-78809390d798.png)  
  
If you have problems launching the tool, look here: https://www.reddit.com/user/lillehavard123/  
[Microsoft Visual C++ 2008 Service Pack 1 Redistributable Package](https://www.microsoft.com/en-us/download/details.aspx?id=26368)  
![image](https://github.com/mriscoc/Ender3V2S1/assets/2745567/f2f0dcf7-3cb4-4e78-9779-3ebf95b71148)

**Comment by daniel-volz** [comment]
Thanks for your reply.
A chinese Software is really hard to use, but i will have a look on it.

**Comment by alexankitty** [comment]
It's been a bit since this has been touched on, but figured I'd reply here since information is scarce.
I created a bunch of icon color mods myself, and documented how to make your own.
You can get to all of that information here. https://github.com/alexankitty/CrealityEnderV2ScreenColors
Hopefully that helps someone else as well.
And thanks to @mriscoc for supplying the key piece to the puzzle.

**Reply by tzheng777** [reply]
Thanks for creating the guide @alexankitty 

With regards to the fonts, I've found that the Zurich font looks decent. However all the truetype font files I've put onto the DACAI screen have some glitching where some pixels in the numbers don't get updated. Looking at the original OEM SimSun font file there's some random blank spaces are various letter spots compared to a normal .ttf file. 

Do you know whether the converter tool processes the .ttf file so that it displays properly?

**Reply by alexankitty** [reply]
It doesn't seem like it does. In my testing I couldn't get it to accept any font.ttf, however I spent more time on getting the icons to work so more research needed. Currently on mine I'm using the droid sans font, and it works mostly except the half the number doesn't seem to refresh correctly when it changes. It could be a monospaced font thing, too, though I'm a little worried about constantly flashing it like this.

**Reply by tzheng777** [reply]
Yeah I think that's what I'm seeing too, some part of the number doesn't get updated  - especially when a double digit number becomes a single digit number.

**Comment by mriscoc** [comment]
https://github.com/mriscoc/Ender3V2S1/discussions/112#discussioncomment-2517616 Updated

**Comment by NetTech2001** [comment]
I am having trouble getting the tool to launch in Windows 10, complaining about the side-by-side configuration. Has anyone run into a similar issue?

**Reply by lillehavard** [reply]
I might be way late, but i just created a post showing a fix to this.
https://www.reddit.com/r/ender3v2/comments/1aq6dji/fix_for_if_you_cant_start_xgus/

I go into more detail in the post, but basically download c++ redistributables 2008 x86 (32-bit) version.
